paasta_tools/cli/cmds/mark_for_deployment.py

```diff
@@ -423,60 +427,21 @@ def paasta_mark_for_deployment(args):
             raise ValueError('Failed to find image in the registry for the following sha %s' % commit)
 
     deploy_info = get_deploy_info(service=service, soa_dir=args.soa_dir)
-    slack_notifier = SlackDeployNotifier(
-        deploy_info=deploy_info, service=service,
-        deploy_group=deploy_group, commit=commit, old_commit=old_git_sha, git_url=args.git_url,
-        auto_rollback=args.auto_rollback,
-    )
 
-    ret = mark_for_deployment(
-        git_url=args.git_url,
-        deploy_group=deploy_group,
+    deploy_process = MarkForDeploymentProcess(
```

Comment:
Please gate this on `if args.auto_rollback`.

I know it is cleaner, but we have no itests, and this code can stand to wait until it has proven itself at least a little bit before we put it on the critical path of deploying all services at yelp.